File: codes/python/p057.py
# 
# Solution to Project Euler problem 57
# Copyright (c) Sankha Saha. All rights reserved.
# 
#  
#   
# 


def compute():
	LIMIT = 1000
	ans = 0
	numer = 0
	denom = 1
	for _ in range(LIMIT):
		numer, denom = denom, denom * 2 + numer
		# Now numer/denom is the i'th (0-based) continued fraction approximation of sqrt(2) - 1
		if len(str(numer + denom)) > len(str(denom)):
			ans += 1
	return str(ans)


if __name__ == "__main__":
	print(compute())

File: codes/Haskell/p057.hs
{- 
 - Solution to Project Euler problem 57
 - Copyright (c) Sankha Saha. All rights reserved.
 - 
 -  
 -   
 -}


limit = 1000
main = putStrLn (show ans)
ans = sum [1 | b <- take limit (contFracSeq 0 1), b]

contFracSeq :: Integer -> Integer -> [Bool]
contFracSeq n d = let
		numer = d
		denom = d * 2 + n
	in (length (show (numer + denom)) > length (show denom)) : (contFracSeq numer denom)
